docs/deployment/security/secrets.mdx

```diff
@@ -0,0 +1,135 @@
+}
+
+struct SecretRequest {
+    // The name of the secret as addressed in the secret store
+    name: String,
+    // The version of the secret
+    version: Option<String>,
+    // Context for the request, including application name
+    context: Context,
+}
+
+struct SecretResponse {
+    pub secret: Option<Secret>,
+    pub error: Option<String>,
+}
+
+struct Secret {
+    pub name: String,
+    pub version: String,
+    pub string_secret: Option<String>,
+    pub binary_secret: Option<Vec<u8>>,
+}
+```
+
+One of the key properties of this API is that sensitive payloads are encrypted in transit over NATS using **xkeys**. [Xkeys are x25519 keypairs that are an extension on top of NATS' nkeys format](https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_callout#encryption). These keys are compatible with the [NaCl](https://nacl.cr.yp.to/) Seal and Open operations, which makes them suitable for general purpose encryption and decryption.
+
+The payload of a request to the `wasmcloud.secrets.$name.get` endpoint _must_ be encrypted using a xkey generated by the host for that specific request and provided as a header on the NATS message. This is to prevent replay attacks and eavesdropping by other NATS clients on the secrets backend. If the payload of a `get` request was sent in the clear, in theory all a malicious client would need to do in order to get a copy of secrets for a component or provider would be to resend the original request. While the signed component or provider JWTs are _not_ private, having a copy of them does demonstrate that the caller has access to the built binary and accessing secrets on their behalf. Encrypting the payload with a single use xkey prevents the need for a nonce or a timestamp since every request to the endpoint has a unique payload.
+
+The `server_xkey` endpoint is used to retrieve the public xkey of the secrets backend so that the host can encrypt and decrypt communication to the backend. Callers should cache this key so that they do not have to make repeated calls to the endpoint before calling get. If the server is unable to decrypt a payload it will return an error indicating that, so clients should make a call to the `server_xkey` endpoint to refresh the server's key and try again.
+
+### Get request payload
+
+Headers:
+
+- `Wasmcloud-Host-Xkey`: String
+
+Body:
+
+This payload is encrypted using the xkey received from the backend's `server_xkey` response as the recipient.
+
+```json
+{
+  "secret_name": "",
+  // Optional
+  "version": "1",
+  "context": {
+    "jwt": "..."
+  }
+}
+```
+
+### Get reponse payload
+
+This payload is encrypted using the xkey provided in the `Wasmcloud-Host-Xkey` header as the recipient.
+
+```json
+{
+  // Will be empty if errored
+  "secret": {
+    "name": "",
+    "version": "",
+    // Only one of the following fields will be set
+    "secret_string": "",
+    "secret_binary": ""
+    }
+    // Omitted on success
+    "error": ""
+}
+```
```

Comment:
```suggestion
```jsonc
{
  // Will be empty if errored
  "secret": {
    "name": "",
    "version": "",
    // Only one of the following fields will be set
    "secret_string": "",
    "secret_binary": ""
  },
  // Omitted on success
  "error": ""
}
```
```